### ISSUE TITLE
Light inside a USE does not exist

### ISSUE THREAD

**Issue by daniel-cyberbotics** [x3dom/x3dom#678]
For example, a PointLight is inside a node with DEF, and a node includes a USE to that node.
The result intuitively expected is to have two lights but x3dom only displays one light.

```
<scene render="true" bboxcenter="0,0,0" bboxsize="-1,-1,-1" pickmode="idBuf" dopickpass="true">
<NavigationInfo headlight="false" type="NONE" typeParams="-0.4,60,0.05,2.8" explorationMode="all" avatarSize="0.25,1.6,0.75" speed="1" transitionTime="1" transitionType="LINEAR "></NavigationInfo>
<Environment shadowExcludeTransparentObjects="true" sortTrans="true" gammaCorrectionDefault="linear" frustumCulling="true" smallFeatureThreshold="1" lowPriorityThreshold="1" minFrameRate="1" maxFrameRate="62.5" userDataFactor="-1" smallFeatureFactor="-1" occlusionVisibilityFactor="-1" lowPriorityFactor="-1" tessellationErrorFactor="-1" SSAOradius="0.7" SSAOamount="0.3" SSAOrandomTextureSize="4" SSAOblurDepthTreshold="1"></Environment>
<WorldInfo id="1" info=""></WorldInfo>
<Viewpoint id="2" orientation="-0.33049289 0.86558641 0.37621086 1.8413467" position="1.1795031 1.7118169 0.21957198" zNear="0.05" fieldOfView="0.785398" centerOfRotation="0,0,0" zFar="-1"></Viewpoint>
<Background id="3" groundColor="" groundAngle="" skyColor="0,0,0" skyAngle="" backUrl="" bottomUrl="" frontUrl="" leftUrl="" rightUrl="" topUrl=""></Background>
<Transform id="4" render="true" bboxCenter="0,0,0" bboxSize="-1,-1,-1" center="0,0,0" translation="0 0 0" rotation="0 1 0 0" scale="1,1,1" scaleOrientation="0,0,0,0">
    <Transform id="5" DEF="AAA" translation="0 -0.07 0" render="true" bboxCenter="0,0,0" bboxSize="-1,-1,-1" center="0,0,0" rotation="0 1 0 0" scale="1,1,1" scaleOrientation="0,0,0,0">
        <PointLight id="6" location="0 0.1 0" color="1,1,1" intensity="1" on="true" shadowMapSize="1024" zNear="-1" zFar="-1" attenuation="1,0,0" radius="100"></PointLight>
    </Transform>
</Transform>
<Transform id="7" translation="-1 0 0" render="true" bboxCenter="0,0,0" bboxSize="-1,-1,-1" center="0,0,0" rotation="0 1 0 0" scale="1,1,1" scaleOrientation="0,0,0,0">
    <Transform id="8" USE="AAA" translation="0 -0.07 0" rotation="0 1 0 0"></Transform>
</Transform>
</scene>
```

![65825328-a5cd-11e6-87ca-e19d09f556fb](https://cloud.githubusercontent.com/assets/22005969/20107488/fde75e82-a5d8-11e6-9c0a-408872cd7eb2.png)

**Comment by andreasplesch** [comment]
Hi, 
the USE TRANSFORM would be an exact copy of the DEF TRANSFORM, so all its attributes are ignored. But it should still work. It would be best if you can share a quick codepen to demonstrate on a live scene.

As an aside, I tried out USEd lights in another web x3d browser, and they work there (https://rawgit.com/andreasplesch/cobweb_dom/master/tests/fanOutRoute.xhtml)

**Comment by stefaniapedrazzi** [comment]
Hi, I could reproduce this issue.
here is a simple jsFiddle:
https://jsfiddle.net/spedrazzi/b5eyeh58/7/

I tried to use directly a USE PointLight and also a USE Transform containing a PointLight and a sphere.
The world contains correctly 2 spheres but only one light source is visible instead of 3.

am I doing something wrong?

**Comment by andreasplesch** [comment]
You are right, there should be three lights but there is only one. I checked the generated shader, it also has only one light. My guess is that the USEd light has the same key/hash as the DEF light and is therefore skipped somewhere. You may be able to track down where this happens.

**Comment by andreasplesch** [comment]
light never gets added to lights node bag. Only lights in the bag are used for rendering.

The next issuI tried to follow what happens with the USE lights, and as a result can say that it will require significant work to allow those. 

The first issue is that the USE node is placed in the scene graph as a direct reference to the def node. Therefore the USEe is that the transform hierarchy only gets tracked for 'drawables' . If USE Viewpoint s work, Lights could adopt their transform approach.

Since there is a sensible workaround there would need to be a convincing use case.

**Comment by stefaniapedrazzi** [comment]
Thank you for checking this issue.

We are automatically creating the X3DOM environment based on VRML/X3D worlds created by our users.
Given that in X3D it is valid to use USE lights, we cannot and don't want to prevent our users to do it even if there is no convincing use case.
But we can automatically convert the USE lights into DEF lights when creating the X3DOM version to workaround this difference from standard X3D.

If you have a documentation of DEF/USE nodes it could maybe be useful to add this limitation.

**Comment by andreasplesch** [comment]
Good idea. You may be able to add a repo wiki page to start collecting these limitations.
